src/config.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Module for getting and setting Prebid configuration.
+ *
+ * Prebid previously defined these properties directly on the global object:
+ * pbjs.logging = true;
+ *
+ * Defining and access properties in this way is now deprecated, but these will
+ * continue to work during a deprecation window.
+ */
+const utils = require('./utils');
+
+const DEFAULT_DEBUG = false;
+const DEFAULT_BIDDER_TIMEOUT = 3000;
+const DEFAULT_PUBLISHER_DOMAIN = window.location.origin;
+const DEFAULT_COOKIESYNC_DELAY = 100;
+const DEFAULT_ENABLE_SEND_ALL_BIDS = false;
+
+const ALL_TOPICS = '*';
+
+let listeners = [];
```

Comment:
I agree that what we have here is workable. Honestly, the question which is really in the back of my mind is "how do we get to an API like @protonate suggested, without it being such a massive project?" Something like:
```
prebid.push(function() {
  prebid.doAuction().then(renderAd);
}
```

As written, this config module can only be used in a project which is driven by global state. That's fine... because the current project is driven by global state.

Using my suggestion, this config module could be used in both systems. 

```
function doAuction() {
  const config = newConfig();
  ...auction impl here...
}
```

If we keep adding new modules which rely on global state, then certain features (like infinite scroll) become more and more difficult to support with time.

If we add modules which don't rely on global state, then we can reuse them out of the box if(/when) we migrate in that direction.